fiftyone/server/routes/embeddings.py

```diff
@@ -64,12 +64,18 @@ def _post_sync(self, data):
             dataset = fosu.load_and_cache_dataset(dataset_name)
```

Comment:
@ritch take a look at `load_and_cache_dataset()` and see if it matches the expected behavior for both singleton and non singleton modes.

Reply:
See my response here: https://github.com/voxel51/fiftyone/pull/6161#discussion_r2219471812